capsules/src/mac.rs

```diff
@@ -0,0 +1,365 @@
+//! Implements 802.15.4 MAC device functionality as an abstraction layer over
```

Comment:
This comment should be more objective, and more independent of the general design.

It should basically answer the following questions:

1. What functionality does this capsule implement? (ieee 802.15.4 MAC layer, with a reference to the spec as well as specifying any choices this implementation uses that are as general as the spec).

2.  How would I use it? Specifically, how do the exposed structs and traits relate to each other? How (more or less) would a user set this up to use it? What dependencies does is have (i.e. the `Radio` trait, and I think this is the only place to mention the raw radio transceiver)?